app/javascript/packages/document-capture/components/acuant-selfie-camera.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useContext, useEffect } from 'react';
+import type { ReactNode } from 'react';
+import AcuantContext from '../context/acuant';
+
+declare global {
+  interface Window {
+    AcuantPassiveLiveness: AcuantPassiveLivenessInterface;
+  }
+}
+
+type AcuantPassiveLivenessStart = (
+  faceCaptureCallback: FaceCaptureCallback,
+  faceDetectionStates: FaceDetectionStates,
+) => void;
+
+interface AcuantPassiveLivenessInterface {
+  /**
+   * Start capture
+   */
+  start: AcuantPassiveLivenessStart;
+  /**
+   * End capture
+   */
+  end: () => void;
+}
+
+interface AcuantSelfieCameraContextProps {
+  /**
+   * Success callback
+   */
+  onImageCaptureSuccess: ({ image }: { image: string }) => void;
+  /**
+   * Failure callback
+   */
+  onImageCaptureFailure: any;
+  /**
+   * Capture open callback, tells the rest of the page
+   * when the fullscreen selfie capture page is open
+   */
+  onImageCaptureOpen: () => void;
+  /**
+   * Capture close callback, tells the rest of the page
+   * when the fullscreen selfie capture page has been closed
+   */
+  onImageCaptureClose: () => void;
+  /**
+   * React children node
+   */
+  children: ReactNode;
+}
+
+interface FaceCaptureCallback {
+  onDetectorInitialized: () => void;
+  onDetection: (text) => void;
+  onOpened: () => void;
+  onClosed: () => void;
+  onError: (error) => void;
+  onPhotoTaken: () => void;
+  onPhotoRetake: () => void;
+  onCaptured: (base64Image: Blob) => void;
+}
+
+interface FaceDetectionStates {
+  FACE_NOT_FOUND: string;
+  TOO_MANY_FACES: string;
+  FACE_ANGLE_TOO_LARGE: string;
+  PROBABILITY_TOO_SMALL: string;
+  FACE_TOO_SMALL: string;
+  FACE_CLOSE_TO_BORDER: string;
+}
+
+function AcuantCamera({
+  onImageCaptureSuccess = () => {},
+  onImageCaptureFailure = () => {},
+  onImageCaptureOpen = () => {},
+  onImageCaptureClose = () => {},
+  children,
+}: AcuantSelfieCameraContextProps) {
+  const { isReady, setIsActive } = useContext(AcuantContext);
+  const faceCaptureCallback: FaceCaptureCallback = {
```

Comment:
I'd be a little hesitant about defining this here and then using it inside the `useEffect`, since technically that makes this a dependency of the `useEffect`, but we wouldn't want to set it as one, since assigning a variable like this in the render path means we're allocating a new object every time this component renders. That may risk a desync between the object and what's initialized in `start`.

Instead, I might suggest inlining the object where we call `start` below? That or use `useMemo` to make sure this object never changes.

Reply:
Agreed. I'm going to move this object into the effect.

I should have included a PR comment here, I waffled about this quite a lot before settling on doing it this way. Here's what I'm seeing:
- `useEffect` must have all any function it uses in the dependencies array.
- We don't respect that rule in `acuant-camera.tsx`.
    - Which means that we have explicitly turned off the linter for this at some point? Or maybe we're using an old version/different linter?
- Even when I move this object into the `useEffect` all of the functions like `onImageCaptureSuccess` and `onImageCaptureFailure` should probably be in the dependency array?

So all of ☝️ led me to the current state. I'll move the object into the effect now and that'll be an improvement.

Reply:
> We don't respect that rule in `acuant-camera.tsx`.
> * Which means that we have explicitly turned off the linter for this at some point? Or maybe we're using an old version/different linter?

Correct that we don't use the linter currently. For what it's worth, the AcuantCamera component has created a lot of difficulty for upgrading to React 18 and it may very well be related to the fact that its dependencies are very leaky. (cc @matthinz)

I seem to recall that we may have considered adding the linter in the past. I can't recall the exact reasoning for not moving forward with it, but I know there can often be edge cases where the rule ends up being more nuisance than it's worth. But instances like this are a good counter-point to that concern.

Reply:
I moved the object into the `useEffect` in this [commit](https://github.com/18F/identity-idp/pull/9580/commits/cb289bb08ee36ffefaa01edf3ed85e320100a1df). Leaving this unresolved because I want to think more about the linting/dependencies before merging.

Reply:
Unfortunately I think I have to defer any linting changes to another ticket (that I'll create shortly).

After some experimentation with eslint and hooks I think:
- useEffect hooks could use cleanup. We should really have that linting rule on.
- It's a significant project to make the hook be in line with the linting rule, even without turning it on.


What I did:
```javascript
// Turned on the hooks rule by, in the .eslint file, changing the extends rule to this
 "extends": ["plugin:@18f/eslint-plugin-identity/recommended", "airbnb", "airbnb/hooks"],
```
```shell
# Then in a terminal run
yarn eslint app/javascript/packages/document-capture/components/acuant-camera.tsx
# You'll get a LOT of output, what we care about here is the hooks dependency error.
```

Reply:
For what it's worth, I've been thinking to move away from Airbnb's ruleset ([related Slack comment](https://gsa-tts.slack.com/archives/C0NGESUN5/p1699457381248629?thread_ts=1699457274.493639&cid=C0NGESUN5)) so if/when we do explore this further, I'd rather just enable the particular rules rather than adopt another subconfiguration from Airbnb.